lib/autsr.gi

```diff
@@ -287,44 +289,155 @@ local c,hom,q,a,b,i,t,int,bad,have,ups,up,new,u,good,abort;
+      a:=Normalizer(G,S);
+      # if the normalizer index is large, there will be many subgroups above
+      if IndexNC(a,S)<200 then
+        worked:=fail; # abort and go to other method
       fi;
-    od;
-    CompletionBar(InfoMorph,3,"SubgroupsAboveBlocks ",false);
+    fi;
+
+    # if prime powers, the primes must have worked
+    a:=Representative(c[i]);
+
+    # no coprime power in earlier class?
+    have:=worked<>fail and
+      ForAll(Filtered([2..Order(a)-1],o->Gcd(o,Order(a))=1),
+        o->PositionProperty(c{[1..i-1]},x->a^o in x)=fail);
 
-    if  good then 
-      # actually found something. Recurse to restart. (As index decreased
-      # by a factor, not much loss.)
-      S:=SubgroupConditionAbove(G,cond,S);
+    pp:=false;
+    if have and not IsPrimeInt(Order(a)) then
+      pp:=SmallestPrimeDivisor(Order(a));
+      a:=a^(Order(a)/pp);
+      have:=worked[PositionProperty(c,x->a in x)];
     fi;
+
+    if have then
+      have:=false;
+      a:=PreImagesRepresentative(hom,Representative(c[i]));
+      b:=PreImage(hom,Normalizer(q,Group(Representative(c[i]))));
+      Info(InfoMorph,3,"Do class ",i," order =",Order(Representative(c[i])),
+            ", len=",Size(c[i])," idx=",
+            IndexNC(G,b));
+      t:=RightTransversal(G,b);
+      cnt:=0;
+      for b in t do
+        new:=a^b;
+        if (pp=false or new^pp in S) and locond(new) then 
+          S:=ClosureGroup(S,new);
+          have:=true;
+          cnt:=cnt+1;
+        fi;
+      od;
+      Info(InfoMorph,3,"found ",cnt,": ",Size(S));
+    fi;
+    if worked<>fail then Add(worked,have);fi;
+    i:=i+1;
+  od;
+
+  if worked=fail then
+    # still need to test
+    Info(InfoMorph,2,"Go to blocks test");
+
+    if Index(G,S)>clim then Error("clim"); fi;
+
+    a:=SmallGeneratingSet(G);
+    if Length(GeneratorsOfGroup(G))>Length(a) then
+      b:=Size(G);
+      G:=Group(a);
+      SetSize(G,b);
+    fi;
+
+    repeat
+      good:=false;
+
+      # try to prove no supergroup works
+      t:=RightTransversal(G,S:noascendingchain); # don't try to be clever in
+      # decomposing transversal, as this could be hard
+      a:=Action(G,t,OnRight); # coset action, don't need homomorphism
+      b:=RepresentativesMinimalBlocks(a,MovedPoints(a));
+      Info(InfoMorph,3,"Above are ",Length(b)," blocks");
+
+      if Length(b)*10>IndexNC(G,S) then
+        # there are too many blocks. Direct test is cheaper!
+        S:=SubgroupProperty(G,locond,S);
+      else
+
+        for i in [1..Length(b)] do
+          CompletionBar(InfoMorph,3,"SubgroupsAboveBlocks ",i/Length(b));
+          c:=First(b[i],x->x>1);
+          if cond(t[c]) then 
+            S:=ClosureGroup(S,t[c]);
+            good:=true;
+          fi;
+        od;
+        CompletionBar(InfoMorph,3,"SubgroupsAboveBlocks ",false);
+      fi;
+    until good=false;
+
+  fi;
+
+
+
+
+  Info(InfoMorph,3,"Did ",tstcnt," tests");
+  return S;
+  Error();
```

Comment:
Yes. This old code is obsolete now (but it wasn't clear initially that it would be completely obsolete). I have put it in comments now  (and will delete it in a while).